packages/components/src/components/hds/flyout/body.ts

```diff
@@ -7,15 +7,16 @@ import Component from '@glimmer/component';
 import { deprecate } from '@ember/debug';
 
 export interface HdsFlyoutBodySignature {
+  Args: never;
```

Comment:
[question] as a pattern, do we prefer to have `Args` completely missing (I think I have done something like this, recently), or `Args: never` like here? which one is more "correct™"?

/cc @alex-ju

Reply:
Or is it because below we need to have a `HdsFlyoutBodySignature['Args']`?

Reply:
> [question] as a pattern, do we prefer to have Args completely missing (I think I have done something like this, recently), or Args: never like here? which one is more "correct™"?

not having `Args` is the established pattern. however, considering [the error we encounter in HCP](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1729279817534249) for this subcomponent, I don't see a way of avoiding `never`.

we could use `never` straight in the constructor, but we have this pattern of using the signature in constructors, so what we have in this PR feels more consistent.

```js
constructor(owner: unknown, args: never) {
```

Reply:
> not having Args is the established pattern

@alex-ju Good. So nothing to change on my side :) 

> however, considering [the error we encounter in HCP](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1729279817534249) for this subcomponent, I don't see a way of avoiding never.

@alex-ju @zamoore since this is a deviation from the pattern, should we add a comment above these `Args: never` are not following the established pattern? (and potentially have also a link to this thread, for future reference)

Reply:
I can add a comment, but I think this isn't as much as a deviation as a new rule in the pattern. If the component utilizes the constructor, but also does not take any args, define the type for args as `never`.